src/screens/MainApp/DetailedDiagnose/components/NewComment/index.js

```diff
@@ -1,40 +1,95 @@
-import React, { useState } from 'react'
-import { View, TextInput } from 'react-native'
-import { PrimaryButton, GrayButton, Description } from '~/components'
+import React, { useState, useEffect } from 'react'
+import { View, TextInput, ActivityIndicator, Keyboard } from 'react-native'
 import { theme } from '~/constants'
-import styles from './styles'
+import SvgButton from '~/components/buttons/SvgButton'
+import arrowUpSvg from '~/assets/images/DetailedDiagnose/arrowUp.svg'
+import styles, { SEND_BUTTON_ICON_HEIGHT, SEND_BUTTON_ICON_WIDTH, INPUT_PADDING_ON_KEYBOARD_OPENED, INPUT_PADDING_STATIC } from './styles'
+
+const SendButton = ({ onPress, disabled, submitting }) => {
+  if (submitting) {
+    return (
+      <View style={styles.sendButton}>
+        <ActivityIndicator color='white' />
+      </View>
+    )
+  } else {
+    return (
+      <SvgButton
+        buttonStyle={styles.sendButton}
+        svg={arrowUpSvg}
+        width={SEND_BUTTON_ICON_WIDTH}
+        height={SEND_BUTTON_ICON_HEIGHT}
+        onPress={onPress}
+        disabled={disabled}
+      />
+    )
+  }
+}
 
 const NewComment = ({ onNewComment }) => {
   const [comment, setComment] = useState('')
   const [inputEnabled, setInputEnabled] = useState(true)
+  const [submitting, setSubmitting] = useState(false)
+  const [commentPadding, setCommentPadding] = useState(INPUT_PADDING_STATIC)
+
+  const sendDisabled = (comment.length === 0)
+
+  useEffect(() => {
+    const keyboardDidShow = () => {
+      setCommentPadding(INPUT_PADDING_ON_KEYBOARD_OPENED)
+    }
+
+    const keyboardDidHide = () => {
+      setCommentPadding(INPUT_PADDING_STATIC)
+    }
+
+    const keyboardDidShowListener = Keyboard.addListener(
+      'keyboardDidShow',
+      keyboardDidShow
+    )
+
+    const keyboardDidHideListener = Keyboard.addListener(
+      'keyboardDidHide',
+      keyboardDidHide
+    )
+
+    const cleanUp = () => {
+      keyboardDidShowListener.remove()
+      keyboardDidHideListener.remove()
+    }
+
+    return cleanUp
```

Comment:
:thinking: I think creating functions to immediately use them creates unnecessary noise. Every should programmer knows what `addListener` means in any context and programming environment, so it should be obvious that the function you're passing as second parameter is the event handler. Also, every programmer using React should know its API. It should be obvious that if you return a function, it's a cleanup function. This is much cleaner and shorter (and in turn, easier to read) IMO:
```javascript
useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      () => setCommentPadding(INPUT_PADDING_ON_KEYBOARD_OPENED)
    )

    const keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      () => setCommentPadding(INPUT_PADDING_STATIC)
    )

    return () => {
      keyboardDidShowListener.remove()
      keyboardDidHideListener.remove()
    }
}, [])
```